mainline.sh

```diff
@@ -31,10 +31,10 @@ shift $((OPTIND-1))
 export CC=clang
 export CXX=clang++
 FUZZ_FLAG="-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
-export CFLAGS="-fsanitize=address"
-export CXXFLAGS="-fsanitize=address -stdlib=libstdc++ $FUZZ_FLAG"
+export CFLAGS="-fsanitize=address,fuzzer-no-link"
```

Comment:
Why are these required?

Reply:
`fuzzer-no-link` as added as the harnesses weren't finding any coverage when running. After adding this, the warning went away.

```
WARNING: no interesting inputs were found so far. Is the code instrumented for coverage?
This may also happen if the target rejected all inputs we tried so far
```

As for linking stdc++, OpenSSL fails to build on my Ubuntu system without linking it:

```
${LDCMD:-clang-17} -pthread -m64 -fno-omit-frame-pointer -g -Wa,--noexecstack -Qunused-arguments -fsanitize=address,fuzzer-no-link -fsanitize=address,fuzzer-no-link -fno-sanitize=alignment -L../../usr/lib -L.   \
        -o fuzz/asn1 \
        fuzz/asn1-bin-asn1.o fuzz/asn1-bin-driver.o \
        fuzz/asn1-bin-fuzz_rand.o \
        -lFuzzingEngine -lssl -lcrypto -ldl -pthread 
/usr/bin/ld: ../../usr/lib/libFuzzingEngine.a(FuzzerIO.cpp.o): in function `fuzzer::FileToVector(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, bool)':
(.text._ZN6fuzzer12FileToVectorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEmb+0x2b): undefined reference to `std::basic_ifstream<char, std::char_traits<char> >::basic_ifstream(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::_Ios_Openmode)'
// many many more lines (...)
```

I'm using clang 17 and the following for the fuzzing engine:
```
lrwxrwxrwx 1 root root 67 Dec 13 15:58 /usr/lib/libFuzzingEngine.a -> /usr/lib/llvm-17/lib/clang/17/lib/linux/libclang_rt.fuzzer-x86_64.a
```

Reply:
I'm just a bit surprised it's needed. Have you tried building the fuzzers with the oss-fuzz images or just locally?

Reply:
building through the oss-fuzz scripts seemed to be fine when I tried, but that goes through a different script (`ossfuzz.sh`). I haven't compared the environments to see what the difference is; it's possible it might be related to the compiler or fuzzing engine versions I'm using.

Reply:
Ok. Happy with this then. If you can fix up the merge conflicts I'll get this merged in.